runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -7590,8 +7611,29 @@ done:;
 		VM_BytecodeAction rc = EXECUTE_BYTECODE;
 		j9object_t obj = *(j9object_t*)_sp;
 		_sp += 1;
+
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+		J9Class * objClass = NULL;
+
+		if (NULL == obj) {
```

Comment:
This is the same with or without valuetypes so it can be taken out of the ifdef

Reply:
same for monitor enter

Reply:
```
if (NULL = obj) {
   ...
} else {
ifdef VT
    if (obj.isValue) {
        ...
    } else 
endif VT
   {
        ... // rest of code
   }
return ...

Reply:
just modified it, I think its fine to fetch the objClass twice since its an exception path. Will need a comment to indicate why that is happening

Reply:
I have modified it. emm, I have never seen a bracket used like this before : ) .

Reply:
not quite, the `{` after the first else should be removed.

And youll have to do `if (J9_IS_J9CLASS_VALUETYPE(J9OBJECT_CLAZZ(_currentThread, obj))`

Reply:
The purpose is to ensure that non-VT remains unchanged

Reply:
please review. Just understand the purpose of such modification, hope this is what you want.